### DISCUSSION TITLE
Unable to transcode videos using VAAPI on AMD GPU

### DISCUSSION THREAD

**Discussion by ancashoria** [immich-app/immich#4846] [General]
### The bug

I have enabled VAAPI hardware acceleration on an Ryzen 4300G APU and it's not able to transcode any videos.

By checking the logs I see only this error:
```
An error occurred while configuring transcoding options: Error: ENOENT: no such file or directory, scandir '/dev/dri'
```

I'm running Immich on TrueNAS using the TrueCharts repository. I will check there for an answer too.
Thank you for making Immich awesome.

### The OS that Immich Server is running on

Truenas Scale 23 (Debian)

### Version of Immich Server

1.84

### Version of Immich Mobile App

1.84

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
none
```

### Your .env content

```Shell
none
```

### Reproduction steps

```bash
Enable hardware transcoding and try to transcode videos
```

**Comment by mertalev** [comment]
Hi! You need to add a device mapping for `/dev/dri` for the container to be able to see it. With Docker Compose, this would be done by adding the below snippet to the `immich-microservices` service:

```
devices:
  - /dev/dri:/dev/dri
```

I'm unfortunately not familiar with how this is done using TrueNAS, but I hope this can point you in the right direction.

**Reply by ancashoria** [reply]
I have added the GPU from the TrueNAS ui, but I guess I should check that somehow.
That being said, Immich is expected to work with AMD gpus? I ask because I see other libraries that had to do extra stuff to get that working: https://github.com/jellyfin/jellyfin/pull/7708

**Reply by mertalev** [reply]
Yes, it should work with AMD GPUs. Jellyfin has end-to-end hardware acceleration, which increases the scope of work quite a lot. Immich, by contrast, only uses hardware acceleration for encoding. This means it doesn't need to handle library or hardware support for decoding and tone-mapping: all acceleration methods can share the same implementation for these. This comes at the cost of higher CPU utilization, though, so I do hope to add support for end-to-end acceleration some day.

**Reply by ancashoria** [reply]
Some updates. The team over at TrueCharts has managed to fix the issue above, but now I'm running into AMD issues. I'm sending the logs over in an effort to identify if it's related to Immich, or the TrueCharts chart.

![image](https://github.com/immich-app/immich/assets/4144110/f1a4c7fe-e61b-4bcb-97d0-97fcab66016a)

**Reply by ancashoria** [reply]
This is probably related to TrueNAS, being based on Debian, perhaps it doesn't have all the bells and whistles.

**Reply by mertalev** [reply]
Could you try mapping `/dev/dri/renderD128:/dev/dri/renderD128` specifically rather than `/dev/dri:/dev/dri`?

**Reply by jharveyb** [reply]
@mertalev that worked for me! I'm using the integrated GPU on a Ryzen 5850U, running debian testing, and verified the offload via nvtop.

This is also what's recommended in the Jellyfin documentation (I'm using the default ffmpeg not the Jellyfin one):

https://jellyfin.org/docs/general/administration/hardware-acceleration/amd/#configure-on-linux-host

Seems worth adding this to the docs?

**Reply by mertalev** [reply]
That isn't necessary anymore since we support specifying the device in the UI now. I believe Jellyfin also does this. But you're absolutely right that we should add documentation for that.
